### PR TITLE
New Feature: Checking number of frames in a list of videos

### PR DESCRIPTION
**Pull request by MSchellenberger**
The feature is not implemented into the filepath yet, but because it has no dependencies on the pipeline code other than utils, it can be used at any position. Maybe we could even have it in the user interface code, or (my preferred solution) have a new class that acts as a data quality ensurer (running some tests on the data before deciding what steps to take -> exclude parts of the recording, solve missing frame issues, etc.)

Additionally, there is now a function in utils that checks whether the items in a list are identical

### CODE REVIEW

File: core/filenumber_checker.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        video_framenumbers = []
+        for elem in videofilepaths:
+            video = Framenumberreader()
+            framenumber = video.run(convert_to_path(elem))
+            video_framenumbers.append(framenumber)
+        if check_if_same(video_framenumbers):
+            print('All videos have the same amount of frames!')
+        else:
+            print('Not all videos have the same number of frames!')
+            pd.DataFrame(video_framenumbers, index = [videofilepaths])
```
Comment (KonKob) [inline core/filenumber_checker.py:24]: What's the intention behind creating this DataFrame?

File: core/utils.py [hunk 2]
```diff
@@ -8,6 +8,12 @@
 import yaml
 
 
+def check_if_same(items: List):
```
Comment (KonKob) [inline core/utils.py:11]: Useful! Could be implemented in triangulation_calibration_module.py to validate that the metadata in all recording/calibration videos is the same (target_fps, recording_date, calibration_index, etc.)

File: core/filenumber_checker.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from typing import List
+from pathlib import Path
+
+import pandas as pd
+from moviepy.video.io.VideoFileClip import VideoFileClip
```
Comment (KonKob) [inline core/filenumber_checker.py:5]: moviepy is a new dependency, that would be needed to run the pipeline. 
it should be installed with DLC? anyways, but maybe we can use imageio instead to stay consistent?

File: core/filenumber_checker.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    """
+    Reads a video, checks & returns the number of frames
+    """
+
+    def run(self, videofilepath: Path) -> int:
+        return(self.read_framenr(self.read_video(videofilepath)))
+
+    def read_video(self, videofilepath: Path) -> None:
+        return VideoFileClip(videofilepath)
```
Comment (KonKob) [inline core/filenumber_checker.py:37]: imageio.v2.get_reader(videofilepath)

File: core/filenumber_checker.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    def read_framenr(self, video) -> int:
+        return video.reader.nframes
```
Comment (KonKob) [inline core/filenumber_checker.py:40]: return video.count_frames()

File: core/filenumber_checker.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
+from typing import List
+from pathlib import Path
+
+import pandas as pd
+from moviepy.video.io.VideoFileClip import VideoFileClip
+
+from utils import check_if_same, convert_to_path
+
+class Framenumberchecker():
```
Comment (KonKob) [inline core/filenumber_checker.py:9]: Could be implemented into the pipeline after synchronisation to make sure that
- synchro worked
- no frame drops in videos

File: core/filenumber_checker.py (hunk 1 above)
Comment (MSchellenberger) [reply core/filenumber_checker.py:24]: showing which videos have different frame numbers. In the next step we could extract the video that is faulty and fix/get around the issue

File: core/filenumber_checker.py (hunk 6 above)
Comment (MSchellenberger) [reply core/filenumber_checker.py:9]: Even before after collecting metadata to ensure that the data quality is good enough

Comment (KonKob) [reply core/filenumber_checker.py]: Before synchronising, videos with different fps will have different numbers of frames. You could also go for duration? (number of frames/fps)?

File: core/filenumber_checker.py (hunk 1 above)
Comment (KonKob) [reply core/filenumber_checker.py]: So, probably you want to save this DataFrame as a variable to make it accessible later on :D

**Comment by KonKob** [comment]
Before merging, please restore your changes in core/angles_and_distances.py and video_synchronization.py, or add some information about what you'd like to change in those files

**Pull request closed without merging**